src/System.Management.Automation/engine/SessionStateContainer.cs

```diff
@@ -1976,16 +1984,43 @@ internal Collection<PSObject> GetChildItems(string[] paths, bool recurse, uint d
                     }
 
                     // Now recurse if it is a container
-                    if (recurse && IsItemContainer(providerInstance, qualifiedPath, context))
+                    if (recurse )
                     {
-                        // Making sure to obey the StopProcessing.
-                        if (context.Stopping)
+                        itemContainer = false;
+                        try
                         {
-                            return;
+                            itemContainer = IsItemContainer(providerInstance, qualifiedPath, context);
+                        }
+                        catch (UnauthorizedAccessException accessException)
+                        {
+                            context.WriteError(new ErrorRecord(accessException, "GetItemUnauthorizedAccessError", ErrorCategory.PermissionDenied, qualifiedPath));
+                        }
+                        catch ( ProviderInvocationException accessException)
+                        {
+                            // if providerinvocationexception is wrapping access denied error, it is ok to not terminate the pipeline
+                            if(accessException.InnerException != null && 
```

Comment:
Done.

---
In reply to: [85202384](https://github.com/PowerShell/PowerShell/pull/2503#discussion_r85202384) [](ancestors = 85202384)